index.bs

```diff
@@ -397,108 +521,242 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/#
+          algorithm returns a {{MediaCapabilitiesInfo}}. The following steps are
+          run:
           <ol>
             <li>
-              If the user agent is able to decode the media represented by
-              <a for=MediaCapabilitiesInfo>configuration</a>, set
-              <a for=MediaCapabilitiesInfo>supported</a> to <code>true</code>.
-              Otherwise set it to <code>false</code>.
+              Let <var>info</var> be a new {{MediaCapabilitiesInfo}} instance.
+              Unless stated otherwise, reading and writing apply to 
+              <var>info</var> for the next steps.
             </li>
             <li>
-              If the user agent is able to decode the media represented by
-              <a for=MediaCapabilitiesInfo>configuration</a> at a pace that
-              allows a smooth playback, set <a
-              for=MediaCapabilitiesInfo>smooth</a> to <code>true</code>.
-              Otherwise set it to <code>false</code>.
+              Set <a for=MediaCapabilitiesInfo>configuration</a> to
+              <var>configuration</var>.
             </li>
             <li>
-              If the user agent is able to decode the media represented by
-              <a for=MediaCapabilitiesInfo>configuration</a> in a power efficient
-              manner, set <a for=MediaCapabilitiesInfo>powerEfficient</a> to
-              <code>true</code>. Otherwise set it to <code>false</code>. The
-              user agent SHOULD NOT take into consideration the current power
-              source in order to determine the decoding power efficiency unless
-              the device's power source has side effects such as enabling
-              different decoding modules.
+              If <a for=MediaCapabilitiesInfo>configuration</a> is of type
+              {{MediaDecodingConfiguration}}, run the following substeps:
+              <ol>
+                <li>
+                  If <code>configuration.keySystemConfiguration</code> is 
+                  <a>present</a>:
+                  <ol>
+                    <li>
+                      Set <a for=MediaCapabilitiesInfo>keySystemAccess</a> to 
+                      the result of running the <a>Check Encrypted Decoding Support</a>
+                      algorithm with <var>configuration</var>. 
+                    </li>
+                    <li>
+                      If <a for=MediaCapabilitiesInfo>keySystemAccess</a> is not
+                      <code>null</code> set 
+                      <a for=MediaCapabilitiesInfo>supported</a> to 
+                      <code>true</code>. Otherwise set it to <code>false</code>. 
+                    </li>
+                  </ol>
+
+                  Otherwise, if the user agent is able to decode the media 
+                  represented by <var>configuration</var>, set 
+                  <a for=MediaCapabilitiesInfo>supported</a> to 
+                  <code>true</code>. Otherwise set it to <code>false</code>. 
```

Comment:
I would suggest doing:
```
</li>
<li>
Otherwise, run the following steps:
<ol>
<li>If the user agent is able to decode the media represented by <var>configuration</var>, set <a for=MediaCapabilitiesInfo>supported</a> to <code>true</code>.</li>
<li>Otherwise, set it to <code>false</code>.</li>
</ol>
</li>
```